pkg/util/clock.go

```diff
@@ -137,6 +142,10 @@ func (f *FakeClock) HasWaiters() bool {
 	return len(f.waiters) > 0
 }
 
+func (f *FakeClock) Sleep(d time.Duration) {
+	f.Step(d)
```

Comment:
I think this is a bit confusing. Typically only the test code changes the clock setting. So I think the default expectation people would have is that Sleep would *block* until the test advances the clock far enough (you could implement it via After).

I guess I don't mind this too much but you should explain in a comment.